internal/v5/resources.go

```diff
@@ -174,3 +135,58 @@ func validResourceName(name string) bool {
 	// TODO we should probably be more restrictive than this.
 	return !strings.Contains(name, "/")
 }
+
+// GET id/meta/resource
+// https://github.com/juju/charmstore/blob/v5/docs/API.md#get-idmetaresources
+func (h *ReqHandler) metaResources(entity *mongodoc.Entity, id *router.ResolvedURL, path string, flags url.Values, req *http.Request) (interface{}, error) {
+	if entity.URL.Series == "bundle" {
+		return nil, nil
+	}
+	ch, err := h.entityChannel(id)
+	if err != nil {
+		return nil, errgo.Mask(err)
+	}
+	resources, err := h.Store.ListResources(entity, ch)
+	if err != nil {
+		return nil, errgo.Mask(err)
+	}
+	results := make([]params.Resource, len(resources))
+	for i, res := range resources {
+		result, err := fromResourceDoc(res, entity.CharmMeta.Resources)
+		if err != nil {
+			return nil, err
+		}
+		results[i] = *result
+	}
+	return results, nil
+}
+
+func fromResourceDoc(doc *mongodoc.Resource, resources map[string]resource.Meta) (*params.Resource, error) {
+	meta, ok := resources[doc.Name]
+	if !ok {
+		// Should never happen.
+		return nil, errgo.Newf("resource %q returned by ListResources but not found in metadata", doc.Name)
+	}
+	r := &params.Resource{
+		Name:        doc.Name,
+		Revision:    -1,
```

Comment:
Pot of curiosity, isn't doc.Revision -1 in the case there is no file? Is this like this just to be more explicit?